pkg/component/etcd/statefulset/values_helper.go

```diff
@@ -119,8 +120,16 @@ func GenerateValues(
 	}
 
 	values.EtcdCommand = getEtcdCommand()
-	values.ReadinessProbeCommand = getReadinessProbeCommand(values)
-	values.LivenessProbCommand = getLivenessProbeCommand(values)
+
+	// Use linearizability for readiness probe so that pod is only considered ready
+	// when it has an active connection to the cluster and the cluster maintains a quorum.
+	values.ReadinessProbeCommand = getProbeCommand(values, linearizable)
+
+	// Use serializability for liveness probe because we are only interested in the health of this particular member.
+	// Those read requests will especially succeed if quorum is lost. Usually it doesn't make sense to also restart
+	// members which are healthy in general, independent of the quorum situation.
+	values.LivenessProbCommand = getProbeCommand(values, serializable)
```

Comment:
This here lead to a case of both livenessProbe and readinessProbe having similar values
```
livenessProbe:
          exec:
            command:
            - /bin/sh
            - -ec
            - ETCDCTL_API=3 etcdctl --endpoints=http://etcd-main-local:2379 get foo
              --consistency=s

readinessProbe:
          exec:
            command:
            - /bin/sh
            - -ec
            - ETCDCTL_API=3 etcdctl --endpoints=http://etcd-main-local:2379 get foo
              --consistency=l
```

This isn't the intention, right?

Reply:
This is the intended behavior. The goal is to get a response from the etcd - in this case via an exec call due to the client cert issue mentioned earlier. `consistency` decides whether the response requires quorum or not - liveness probe doesn't require the member to have quorum, whereas readiness probe requires a quorate cluster. The readiness probe is essentially, in function, similar to a http call to the etcd member's `/health` endpoint, since it too returns true for a quorate cluster.